dotaservice/lua/hero_selection.lua

```diff
@@ -16,25 +61,67 @@ function GetBotNames ()
     }
 end
 
-local bot_heroes = {"npc_dota_hero_nevermore"}
+local function mode1v1()
+    local ids = GetTeamPlayers(GetTeam())
+    for i,v in pairs(ids) do
+        if IsPlayerBot(v) and IsPlayerInHeroSelectionControl(v) and GetSelectedHeroName(v) == "" then
+            _, strHero = get_new_selections( i, "pick" )
+            SelectHero( v, strHero );
+            return
+        end
+    end
+end
+
+local function modeAP()
+    local ids = GetTeamPlayers(GetTeam())
+end
+
+local function modeCM()
+    local ids = GetTeamPlayers(GetTeam())
+end
 
 function Think()
 	-- This gets called (every server tick AND until all heroes are picked).
 	-- This needs to gets called at least once if there is no human.
-    local ids = GetTeamPlayers(GetTeam())
-    for i,v in pairs(ids) do
-        -- If the human is in the unassigned slot, the radiant bots start at v = 2
-        -- If the human is in the radiant coach slot, the radiant bots start at v = 2
-        -- If the human is in the first radiant slot, the radiant bots start at v = 0
-        -- If the human is in the second radiant slot, the radiant bots start at v = 1
-        -- If the human is in the third radiant slot, the radiant bots start at v = 2
-		if IsPlayerBot(v) and IsPlayerInHeroSelectionControl(v) then
-        -- if i == 1 and GetTeam() == TEAM_RADIANT then
-            if i == 1 then
-                SelectHero( v, "npc_dota_hero_nevermore" );
-            else
-                SelectHero( v, "npc_dota_hero_sniper" );
-            end
-		end
-	end
+
+    -- NOTE: other remaining game modes of possible interest
+    -- GAMEMODE_ARDM -- All Random Death Match
+    -- GAMEMODE_ABILITY_DRAFT -- Ability Draft
+
+    local game_mode = GetGameMode()
+    -- All Pick
+    if game_mode == GAMEMODE_AP then
+        modeAP()
+    -- Captain's Mode
+    elseif game_mode == GAMEMODE_CM then
+        modeCM()
+    -- 1v1 Mid
+    elseif game_mode == GAMEMODE_1V1MID then
+        mode1v1()
+    else
+        print("[ERROR] UNKNOWN GAME MODE: ", game_mode)
+        do return end
+    end
+end
+
+-- Function below sets the lane assignments for default bots
+-- Obviously, our own agents will do what they belive best
+function UpdateLaneAssignments()

```

Comment:
Why do we care about lane assignments?

Reply:
That's only if we plan on playing with the default bots. It forces the default bots to not try and assume the mid-lane is empty for example when all your bot (or you as a human) did was go get the off-lane rune at spawn time.

If we are just doing 5v5 full-agent (or 1v1 self-play) than it doesn't matter to us at all.

Reply:
Yeah i would love to have 1v1 against default bot working to validate my self play. I actually want that ASAP.

Reply:
Btw can you add what you just said as comment